# Statistics Drill with Python

In [1]:
import pandas as pd
import statistics
import numpy as np

df = pd.DataFrame()
df['age'] = [14, 12, 11, 10, 8, 8, 6]
print("mean: ", np.mean(df['age']))
print("median: ", statistics.median(df['age']))
print("mode: ", statistics.mode(df['age']))
print("variance: ", np.var(df['age']))
print("standard deviation: ", np.std(df['age']))
print("standard error: ", np.std(df['age'])/ np.sqrt(len(df['age'])))

mean:  9.857142857142858
median:  10
mode:  8
variance:  6.408163265306122
standard deviation:  2.531435020952764
standard error:  0.9567925036515135


In [2]:
#age2 is the updated list with a birthday from age 6 to 7. 
df2 = pd. DataFrame()
df2['age2'] = [14, 12, 11, 10, 8, 8, 7]
print(df2.describe())
print("variance: ", df2['age2'].var())
print("standard error: ",np.std(df2['age2']) / np.sqrt(len(df2['age2'])))

            age2
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000
variance:  6.333333333333333
standard error:  0.8806305718527109


In [3]:
#age3 is the updated list with one less child age 8 and one more child age 1. 
df3 = pd.DataFrame()
df3['age3'] = [14, 12, 11, 10, 8, 7, 1]
print(df3.describe())
print("variance: ", df3['age3'].var())
print("standard error: ", np.std(df3['age3']) / np.sqrt(len(df3['age3'])))

            age3
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000
variance:  18.0
standard error:  1.4846149779161806


In [4]:
dffans = pd.DataFrame()
dffans['fans'] = [20, 23, 17, 5]
print(dffans.describe())
print("variance: ", dffans['fans'].var())
print("standard error: ", np.std(dffans['fans'] ,ddof=1) / np.sqrt(len(dffans['fans'])))

            fans
count   4.000000
mean   16.250000
std     7.889867
min     5.000000
25%    14.000000
50%    18.500000
75%    20.750000
max    23.000000
variance:  62.25
standard error:  3.944933459514875


# Statistics Drill by Hand


1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? 

Greg - 14
Marcia - 12
Peter - 11
Jan - 10
Bobby - 8
Oliver - 8
Cindy - 6

mean - 9.857
median - 10
mode - 8


2. What are the variance, standard deviation, and standard error?

variance - 7.141
standard deviation - 2.672
standard error - 1.01


3. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

I would pick the median and standard deviation to describe this data because while the median and mean are both very close, I think with the kind of data (age) we have, it makes more sense to use Jan's age as an approximation of the children's ages. I would use standard deviation because it helps to visualize how the ages deviate from the center.

4. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

Greg - 14
Marcia - 12
Peter - 11
Jan - 10
Bobby - 8
Oliver - 8
Cindy - 7

mean - 10
median - 10
mode - 8

variance - 6.33
standard deviation - 2.512
standard error - 0.951

The mean has changed. Variance, standard deviation, and standard error have all decreased.

5. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

Greg - 14
Marcia - 12
Peter - 11
Jan - 10
Bobby - 8
Cindy - 7
Jessica - 1

mean - 9
median - 10
mode - none

variance - 17.33
standard deviation - 4.163
standard error - 1.574

Yes, I would choose the mean and variance to describe the data. Since one of the numbers are very low, I think the mean does a better job of including that information. The variance (and standard deviation) is also very high which seems significant. 

6. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. 

The answers were: 
TV Guide 20% fans 
Entertainment Weekly 23% fans 
Pop Culture Today 17% fans 
SciPhi Phanatic 5% fans. 

7. Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

We are working with a sample of the population. I will calculate the measures of central tendancy. 

mean - 16.25
median - 20
mode - none

variance - 63.25
standard deviation - 7.89
standard error - 3.945

Based on these results, I don't think I can make a good estimate. But I do not think I have any choice currently but to guess that 16.25 of our adult population were fans of the show. 